src/components/app/DragAndDrop.js

```diff
@@ -0,0 +1,104 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import classNames from 'classnames';
+import { retrieveProfileFromFile } from '../../actions/receive-profile';
+import type { ConnectedProps } from '../../utils/connect';
+
+function _dragPreventDefault(event: DragEvent) {
+  event.preventDefault();
+}
+
+type OwnDragAndDropProps = {|
```

Comment:
OwnDragAndDropProps, DispatchDragAndDropProps, etc. can all be simplified to just `OwnProps`, `DispatchProps`, etc since there is no need to disambiguate the props from other props on the patch. It should make them a bit easier to read and understand when scanning the code.